business/istio_config.go

```diff
@@ -244,10 +296,11 @@ func (in *IstioConfigService) GetIstioConfigList(ctx context.Context, criteria I
 		if criteria.Include(kubernetes.DestinationRules) {
 			var err error
 			// Check if namespace is cached
+			// @TODO better way to check IsResourceCached per kubeCache
 			if IsResourceCached(criteria.Namespace, kubernetes.DestinationRules) {
-				istioConfigList.DestinationRules, err = in.cache.GetDestinationRules(criteria.Namespace, criteria.LabelSelector)
+				istioConfigList.DestinationRules, err = kubeCache.GetDestinationRules(criteria.Namespace, criteria.LabelSelector)
 			} else {
-				drl, e := in.k8s.Istio().NetworkingV1beta1().DestinationRules(criteria.Namespace).List(ctx, listOpts)
+				drl, e := in.userClients[cluster].Istio().NetworkingV1beta1().DestinationRules(criteria.Namespace).List(ctx, listOpts)
```

Comment:
Here, I'm not sure we need to do this, because as we are using informers/listers to fetch kubernetes resources, we need to use them to fetch things, the GetDestinationRules method from the cache, use the destionationRuleLister, and if I understand them correctly, they are subscribe to events that kubernetes send after there is an update/creation,etc, and there is an internal cache store every resource (the client-go library keeps them in sync with kubernetes).

Reply:
Will not it be just an unreachable code in the when cache is always available?
But if by some reason cache is not available, it still can call directly the k8s client.
This is more historical , keeping the existing logic.

Reply:
IsResourceCached also checks that the type is cached. You can specify which istio types you want cached here: https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.kubernetes_config.cache_istio_types. Not sure why you would ever not cache it and IMO we should remove that config option but it's still valid today.

Reply:
After seeing this in a few other places I'm wondering if instead of checking every occurrence in the business layer if it'd be better to check this in the handlers. It's probably worth also checking in the business layers since we really don't want to cause a panic.